### PR TITLE
[test] [admin] branch-2.9: fixed eror test case at AdminApiTest2

### LINKED ISSUE #14863: [Test] org.apache.pulsar.broker.admin.AdminApi2Test in branch-2.9 have multiple failed tests
**Describe the bug**

**To Reproduce**
1. Build from branch-2.9
2. Run all tests of org.apache.pulsar.broker.admin.AdminApi2Test
3. See multiple failed tests

**Expected behavior**
The tests should get passed
The CI should check this test in branch-2.9

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by poorbarcode**
### Contribution Checklist
Fixes #14863

### Motivation
see #14863

### Modifications
- `Namespace.Base.internalGetAntiAffinityNamespaces` fixed NullPointerException
- `NamespacesBase.internalDeleteNamespaceForcefully` Forcibly delete the order of `topics` and `bundles`, becase after deleted bundles, the topic will be fenced, then the operation that delete topic will failure
- fixed `AdminApi2Test` problem test case
  - brokerNamespaceIsolationPoliciesUpdateOnTime: change immutable list to ArrayList
  - testDistinguishTopicTypeWhenForceDeleteNamespace: fixed outdated test case.  `NamespaceService.getFullListOfTopics` at last version not supports dirty data any more.
  - testMaxTopicsPerNamespace:  fixed outdated test case. 1. Adding asynchronous tests; 2. At last version not allow create system topic by hand-operation

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (don't know)

### Documentation

Need to update docs? 

- [ ] `doc-required` 
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 1]
```diff
@@ -458,8 +461,12 @@ protected void internalDeleteNamespaceForcefully(AsyncResponse asyncResponse, bo
             for (NamespaceBundle bundle : bundles.getBundles()) {
                 // check if the bundle is owned by any broker, if not then we do not need to delete the bundle
                 if (pulsar().getNamespaceService().getOwner(bundle).isPresent()) {
-                    futures.add(pulsar().getAdminClient().namespaces()
-                            .deleteNamespaceBundleAsync(namespaceName.toString(), bundle.getBundleRange(), true));
+                    CompletableFuture<Void> futureForDeleteBundle = FutureUtil.waitForAll(futuresForDeleteTopics)
```
Comment (poorbarcode) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:464]: Forcibly delete the order of topics and bundles, becase after deleted bundles, the topic will be fenced, then the operation that delete topic will failure

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiTest2.java [hunk 2]
```diff
@@ -1027,9 +1028,11 @@ public void brokerNamespaceIsolationPoliciesUpdateOnTime() throws Exception {
         parameters1.put("min_limit", "1");
         parameters1.put("usage_threshold", "100");
 
+        ArrayList<String> primary = new ArrayList<>(2);
+        primary.add(brokerName + ".*");
         NamespaceIsolationData nsPolicyData1 = NamespaceIsolationData.builder()
                 .namespaces(Collections.singletonList(ns1Name))
-                .primary(Collections.singletonList(brokerName + ".*"))
```
Comment (poorbarcode) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiTest2.java:1032]: this collection will execute `clear` & `add`, so it should not be immutable

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiTest2.java [hunk 3]
```diff
@@ -1613,40 +1616,70 @@ public void testForceDeleteNamespace() throws Exception {
...
+        // insert an invalid topic name at namespace02
         pulsar.getLocalMetadataStore().put(
-                "/managed-ledgers/" + exNs + "/persistent/", "".getBytes(), Optional.empty()).join();
+                "/managed-ledgers/" + namespace02 + "/persistent/", "".getBytes(), Optional.empty())
+                .join();
 
-        List<String> topics = pulsar.getNamespaceService().getFullListOfTopics(NamespaceName.get(ns)).get();
-        List<String> exTopics = pulsar.getNamespaceService().getFullListOfTopics(NamespaceName.get(exNs)).get();
+        // namespace02: ensure fail due to invalid topic when execute "list topic"
+        try {
```
Comment (poorbarcode) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiTest2.java:1642]: this api at not supports dirty data any more.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiTest2.java [hunk 4]
```diff
@@ -1745,39 +1780,6 @@ public void testMaxTopicsPerNamespace() throws Exception {
             admin.topics().createNonPartitionedTopic(topic + i + i);
         }
 
-        // check first create normal topic, then system topics, unlimited even setMaxTopicsPerNamespace
```
Comment (poorbarcode) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiTest2.java:1748]: At last version not allow create system topic by hand-operation

**Review by eolivelli — CHANGES_REQUESTED** [review]
we are not fixing a test, but I see changes in pulsar-broker.
Is this a patch ported from master branch ?
if not...do we have a similar patch in master branch ? it is better to cherry-pick and do not let branches diverge.

**Comment by poorbarcode** [comment]
> we are not fixing a test, but I see changes in pulsar-broker. Is this a patch ported from master branch ? if not...do we have a similar patch in master branch ? it is better to cherry-pick and do not let branches diverge.

from branch-2.9

**Comment by poorbarcode** [comment]
> it is better to cherry-pick and do not let branches diverge.

Good idea.  Should I split it to two submit ?

**Comment by eolivelli** [comment]
I am sorry I cannot understand.
This patch it targeting branch-2.9, it cannot be a patch ported "from" branch-2.9.

if the problem is also on master branch please open a PR against master branch

**Comment by poorbarcode** [comment]
> This patch it targeting branch-2.9, it cannot be a patch ported "from" branch-2.9.

I am sorry I cannot understand. Could you elaborate on that

**Comment by poorbarcode** [comment]
> if the problem is also on master branch please open a PR against master branch

I agress

**Comment by eolivelli** [comment]
> I am sorry I cannot understand. Could you elaborate on that
see here:
image

btw please open the PR against master branch, we will cherry-pick to active branches.
thanks.

**Comment by poorbarcode** [comment]
> > I am sorry I cannot understand. Could you elaborate on that
> 
> see here: image
> 
> btw please open the PR against master branch, we will cherry-pick to active branches. thanks.

Got it, Thanks

**Pull request closed without merging**